### PR TITLE
GraphQL: remove deprecated resultCount field

### PR DESCRIPTION
**Pull request by camdencheek**
This field has been deprecated [since 2019](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/commit/b90d1f1ed32f0751c4a2a5ee2f71da5db15080fb?visible=3), and it's still being used by some internal clients. This replaces all the remaining references to `resultCount` with `matchCount` and removes the field. 

## Test plan

Covered by integration tests. Also, marked deprecated for over 2 years.

### CODE REVIEW

File: doc/api/graphql/index.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ The entire API can be used via `curl` (or any HTTP library), just the same as an
   We want line breaks for readability, but backslashes to escape them do not work cross-platform.
   This uses line breaks that are rendered but not copy-pasted to the clipboard.
 -->
-<pre class="pre-wrap"><code>curl<span class="virtual-br"></span> -H 'Authorization: token YOUR_TOKEN'<span class="virtual-br"></span> -d '{"query":"query($query: String!) { search(query: $query) { results { resultCount } } }","variables":{"query":"Router"}}'<span class="virtual-br"></span> https://sourcegraph.com/.api/graphql</code></pre>
+<pre class="pre-wrap"><code>curl<span class="virtual-br"></span> -H 'Authorization: token YOUR_TOKEN'<span class="virtual-br"></span> -d '{"query":"query($query: String!) { search(query: $query) { results { matchCount } } }","variables":{"query":"Router"}}'<span class="virtual-br"></span> https://sourcegraph.com/.api/graphql</code></pre>
```
Comment (keegancsmith) [inline doc/api/graphql/index.md:89]: ouch used in the example documentation. I wonder how many people may be using this. Either way you are right, been deprecated for a long time.

**Review by keegancsmith — APPROVED** [review]

File: doc/api/graphql/index.md (hunk 1 above)
Comment (camdencheek) [reply doc/api/graphql/index.md:89]: Yeah 😕, that part is unfortunate. Ideally, we'd have removed all references to it along with deprecation, but it seems there were enough examples of `resultCount` lying around that we were still using it in a few different places. I _think_ the best solution here is to just rip off the bandaid, but 🤷

**Comment by tw** [comment]
@camdencheek sorry to be a pain, but I have a feeling this may have broken https://github.com/sourcegraph/src-cli

I am experiencing the following GraphQL error this morning, running `src search 'hello'`

```
{
  "locations": [
    {
      "column": 5,
      "line": 112
    }
  ],
  "message": "Cannot query field \"resultCount\" on type \"SearchResults\"."
}
```

Having looked, I can still see a reference here: https://github.com/sourcegraph/src-cli/blob/dcb9b8cc496e81bb99f3fe85cc34d9fdbd84f3d1/cmd/src/search.go#L239

I'm unlikely to have time today to raise a PR, but I'll see if I get time over the weekend.

**Comment by camdencheek** [comment]
> sorry to be a pain

Not a pain at all! Thank you for reporting this! I thought I checked src-cli, but you're totally right that I missed a reference. I'll go ahead and push a fix and get a release out. Though it might be best to revert this PR as well if this breaks all src-cli for all people who don't immediately update to the latest version. Thoughts @keegancsmith?

**Comment by camdencheek** [comment]
@tw I just opened [a PR](https://github.com/sourcegraph/sourcegraph/pull/31795) to add that field back so it doesn't break existing `src` installs. Until that gets pushed out, a workaround is to use `src` with the stream API with `src search -stream 'hello'`. Thanks again for the bug report 👍

**Comment by keegancsmith** [comment]
agreed, lets revert

**Pull request merged**